store/views.py

```diff
@@ -1,82 +1,136 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
 from django.shortcuts import get_object_or_404
 from store.models import *
 from django.contrib.auth.decorators import login_required
 from django.http import JsonResponse
 from django.views.decorators.csrf import csrf_exempt
+from django.utils import timezone
+from django.db.models import Avg
+from django.core.exceptions import ObjectDoesNotExist
 
 # Create your views here.
 
+
 def index(request):
     return render(request, 'store/index.html')
 
+
+@csrf_exempt
 def bookDetailView(request, bid):
-    template_name = 'store/book_detail.html'
+    book = get_object_or_404(Book, pk=bid)
+    book.rating = BookRating.objects.filter(
+        book=book).aggregate(rating=Avg('rating'))['rating']
+    book.save()
+
     context = {
-        'book': None, # set this to an instance of the required book
-        'num_available': None, # set this to the number of copies of the book available, or 0 if the book isn't available
+        'book': None,
+        'num_available': None,
     }
-    # START YOUR CODE HERE
-    
-    
+    if request.user.is_authenticated:
+        try:
+            context['user_review'] = BookRating.objects.get(
+                book=book, username=request.user).rating
+        except ObjectDoesNotExist:
+            context['user_review'] = None
+
+    context['num_available'] = BookCopy.objects.filter(
+        book=book, status=True).count()
+    context['book'] = book
+    template_name = 'store/book_detail.html'
+
     return render(request, template_name, context=context)
 
+@csrf_exempt
+@login_required
+def book_rating(request, bid):
+    book = get_object_or_404(Book, pk=bid)
+
+    user = request.user
+    rating = request.POST['rating']
```

Comment:
This may fail with invalid book ID given in POST request, and would lead to server error. Expected behavior is to inform user with Not found (404) error.